internal/services/synapse/synapse_spark_pool_resource.go

```diff
@@ -213,7 +213,7 @@ func resourceSynapseSparkPool() *pluginsdk.Resource {
 			"spark_version": {
 				Type:     pluginsdk.TypeString,
 				Optional: true,
-				Default:  "2.4", // TODO: make this field Required in 4.0 since the default value changes over time
+				Default:  "3.4", // TODO: make this field Required in 4.0 since the default value changes over time
```

Comment:
Can you make the necessary 4.0 feature flagged changes here please?

Reply:
working on it and running the tests

Reply:
updated, add `if !features.FourPointOh() ` for  `spark_version`